packages/components/src/components/post-collapsible/post-collapsible.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Element, h, Method, Prop, State } from '@stencil/core';
+import { checkValidHTMLTag, getElementHeight, onTransitionEnd } from '../../utils';
+
+let nextId = 0;
+
+@Component({
+  tag: 'post-collapsible',
+  styleUrl: 'post-collapsible.scss',
+  shadow: true,
+})
+export class PostCollapsible {
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop({ mutable: true }) collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the collapsible header within the headings structure.
+   */
+  @Prop() headingLevel?: number = 2;
+
+  @State() collapseClasses: string;
+  @State() collapseHeight: string | null = null;
+
+  @Element() host: HTMLElement;
+
+  hasHeader: boolean;
+  headingTag: string | undefined;
+  collapsibleId: string;
+  collapsibleElement: HTMLElement;
+
+  componentWillLoad() {
+    this.hasHeader = !!this.host.querySelector('[slot="header"]');
+    if (!this.hasHeader) {
+      console.warn('Be sure to bind the post-collapsible to its control using aria-controls and aria-expanded attributes. More information here: https://getbootstrap.com/docs/5.2/components/collapse/#accessibility');
+    }
+
+    this.headingTag = checkValidHTMLTag(`h${this.headingLevel}`, 'The post-collapsible element requires a heading level between 1 and 6.');
+
+    this.collapsibleId = this.host.id || `post-collapsible-${nextId++}`;
+    this.collapseClasses = this.collapsed ? 'collapse' : 'collapse show';
```

Comment:
I would prefer to have a private function to set the collapse classes, because you need to this multiple times (line 41 and 60).
If we ever need to update the classes, we can do it only at one place.

Reply:
I just refactored with a private method (see [this commit](https://github.com/swisspost/design-system/pull/683/commits/fd299a3944e4284ab8dfbf72b29cff1a9c07a04b)) but it seems less readable, don't you think?

Reply:
Your right! But if you use the class only to return the correct value and asign this to the variable it should become much clearer:

```javascript
this.collapseClasses = this.getCollapseClasses();

// more lines of code

private getCollapseClasses() {
  return this.collapsed ? 'collapse' : 'collapse show';
}
```